core/field_variable.js

```diff
@@ -143,11 +171,16 @@ Blockly.FieldVariable.dropdownCreate = function() {
   if (this.sourceBlock_) {
     workspace = this.sourceBlock_.workspace;
   }
-
   if (workspace) {
+    var variableTypes = this.getVariableTypes_();
+    var variableModelList = [];
     // Get a copy of the list, so that adding rename and new variable options
     // doesn't modify the workspace's list.
-    variableModelList = workspace.getVariablesOfType('');
+    for (var i = 0, variableType; i < variableTypes.length; i++) {
```

Comment:
You are right that that is what we usually do :).
This doesn't work if variableTypes = [""]. Because then going through the first loop, it exits immediately because "" == False.

Let me know if you want me to initialize variableType in another place.